### PR TITLE
Add a option to mount a sub-directory

### LINKED ISSUE #549: Mount subdirectory of a volume
Requested by https://github.com/juicedata/juicefs/discussions/548

### PR DESCRIPTION
**Pull request by davies**
closes #549

### CODE REVIEW

File: pkg/meta/redis.go [hunk 1]
```diff
@@ -164,7 +165,38 @@ func newRedisMeta(url string, conf *Config) (Meta, error) {
 	}
 
 	m.checkServerConfig()
-	return m, nil
+	m.root = 1
```
Comment (Sandy4999) [inline pkg/meta/redis.go:168]: delete this line

File: pkg/meta/redis.go [hunk 2]
```diff
@@ -164,7 +165,38 @@ func newRedisMeta(url string, conf *Config) (Meta, error) {
...
+	m.root, err = lookupSubdir(m, conf.Subdir)
+	return m, err
+}
+
+func lookupSubdir(m Meta, subdir string) (Ino, error) {
```
Comment (Sandy4999) [inline pkg/meta/redis.go:173]: move this function to interface.go?

File: pkg/meta/sql.go [hunk 3]
```diff
@@ -182,7 +183,16 @@ func newSQLMeta(driver, dsn string, conf *Config) (*dbMeta, error) {
 			callbacks: make(map[uint32]MsgCallback),
 		},
 	}
-	return m, nil
+	m.root = 1
```
Comment (Sandy4999) [inline pkg/meta/sql.go:186]: delete this line

File: pkg/meta/redis.go [hunk 4]
```diff
@@ -3027,6 +3076,7 @@ func (m *redisMeta) LoadMeta(buf []byte) error {
 		return err
 	}
 
+	dm.FSTree.Attr.Inode = 1
```
Comment (Sandy4999) [inline pkg/meta/redis.go:3079]: maybe also reset mode, uid & gid? i.e. mode=0777, uid=gid=0

File: pkg/meta/sql.go [hunk 5]
```diff
@@ -2587,6 +2614,7 @@ func (m *dbMeta) LoadMeta(buf []byte) error {
 		return err
 	}
 
+	dm.FSTree.Attr.Inode = 1
```
Comment (Sandy4999) [inline pkg/meta/sql.go:2617]: same here

File: pkg/meta/redis.go (hunk 1 above)
Comment (davies) [reply pkg/meta/redis.go:168]: this one is needed, otherwise the lookupSubdir() will fail

File: pkg/meta/redis.go (hunk 2 above)
Comment (davies) [reply pkg/meta/redis.go:173]: Let's keep it here, we may put these together as  `struct defaultMeta{}`

File: pkg/meta/redis.go (hunk 4 above)
Comment (davies) [reply pkg/meta/redis.go:3079]: The attribute of root could be change to anything, so we can't reset it.

**Review by Sandy4999 — APPROVED** [review]

**Pull request merged**